guides/source/threading_and_code_execution.md

```diff
@@ -106,60 +183,113 @@ ensure
+application:
+
+```ruby
+def perform_task_with_record(record)
+  Rails.application.executor.wrap do
+    record.save! # Executes before_save, after_save, etc.
+  end
+end
+```
+
+Or enqueuing or performing a background job within the application:
+
+```ruby
+def enqueue_background_job(job_class, *args)
+  Rails.application.executor.wrap do
+    job_class.perform_later(*args)
+  end
+end
+```
+
+These are just a few of many possible use cases, including rendering views or
+templates, broadcasting via [`Action Cable`](action_cable_overview.html) or
+using [`Rails.cache`](caching_with_rails.html).
+
+### The Reloader
+
+Like the Executor, the
+[Reloader](https://api.rubyonrails.org/classes/ActiveSupport/Reloader.html) also
+wraps application code. The Reloader is only suitable where a long-running
+framework-level process repeatedly calls into application code, such as for a
+web server or job queue.
+
+NOTE: Rails automatically wraps web requests and Active Job workers, so you'll
+rarely need to invoke the Reloader for yourself. Always consider whether the
+Executor is a better fit for your use case.
+
+If the Executor is not already active on the current thread, the Reloader will
+invoke it for you, so you only need to call one. This also guarantees that
+everything the Reloader does, including all its callback executions, occurs
+wrapped inside the Executor.
 
 ```ruby
 Rails.application.reloader.wrap do
   # call application code here
 end
 ```
 
-The Reloader is only suitable where a long-running framework-level process
-repeatedly calls into application code, such as for a web server or job queue.
-Rails automatically wraps web requests and Active Job workers, so you'll rarely
-need to invoke the Reloader for yourself. Always consider whether the Executor
-is a better fit for your use case.
-
-### Callbacks
+#### Callbacks
 
 Before entering the wrapped block, the Reloader will check whether the running
-application needs to be reloaded -- for example, because a model's source file has
-been modified. If it determines a reload is required, it will wait until it's
-safe, and then do so, before continuing. When the application is configured to
-always reload regardless of whether any changes are detected, the reload is
-instead performed at the end of the block.
+application needs to be reloaded (because a model's source file has been
+modified, for example). If it determines a reload is required, it will wait
+until it's safe, and then do so, before continuing. When the application is
+configured to always reload regardless of whether any changes are detected, the
+reload is instead performed at the end of the block.
 
 The Reloader also provides `to_run` and `to_complete` callbacks; they are
 invoked at the same points as those of the Executor, but only when the current
 execution has initiated an application reload. When no reload is deemed
 necessary, the Reloader will invoke the wrapped block with no other callbacks.
 
```

Comment:
It would be nice to show an code example here as well:
````suggestion

```ruby
Rails.application.reloader.to_run do
  # call reloading code here
end
```

````